### ISSUE TITLE
crashing / misbehavior in rescue vehicle simulation 

### ISSUE THREAD

**Issue by thardes2** [eclipse/sumo#9310]
We are currently evaluating various options to implement a rescue lane for emergency vehicles, fire brigade vehicles, etc..
Our approach uses sublanes and the Opposite-Direction-Driving.

However, several issues arise:
- Not all emergency vehicles use the lane
- Vehicle collisions occur. This seems to be related to the sublanes
- The sumo-gui sometimes crashes with segfaults (seems to be related to emergency vehicles with blue light) -- The flows need to be included in the example code (intersection.rou.xml)
- Right-of-way rules are still applied when turning, i.e. when turning left, the driver waits until the intersecting lane is clear

It looks like we are overlooking some important configurations and would appreciate the crucial hint.

---
*If possible, upload an example that shows your problem.*
[exampleSetup.zip](https://github.com/eclipse/sumo/files/7294515/exampleSetup.zip)

**SUMO-version:**
1.10.0
**operating system:**
MacOS, Debian 10, Ubuntu 20.04.3 LTS

**Comment by heinovski** [comment]
To be more specific, when using the sublane model, there are collisions between regular vehicles as well as with emergency vehicles, mostly in lateral direction. This happens especially on the junction but seemingly also on regular lanes.

**Comment by heinovski** [comment]
> > It looks like we are overlooking some important configurations and would appreciate the crucial hint.
> 
> No. You are using a combination of features that has seen few tests (blue light device with opposite direction driving) and are therefore the first to discover slumbering bugs.

Can you have a quick look at the configuration parameters of the emergency vehicles and whether they make sense? We figured that some parts of the unexpected behavior might come from incorrect parameter values.

**Comment by heinovski** [comment]
Regarding collisions: When you enable (only) the 4 car flows (`intersection.rou.xml`), set the probability of each flow to `0.25` (our original values), and remove the first traffic light phase (`intersection.net.xml`), which is only red, we get with the current main branch (ffb35d6d2fa43fd11c951bf2ad2b182c8e4f2428):
```
 Vehicle 'from_west.29'; junction collision with vehicle 'from_east.37', lane=':junc4_17_1', gap=-1.00', latGap=0.00, time=246.00 stage=move.
```

**Comment by namdre** [comment]
> Can you have a quick look at the configuration parameters of the emergency vehicles and whether they make sense? We figured that some parts of the unexpected behavior might come from incorrect parameter values.

The parameters are ok.

**Comment by namdre** [comment]
The junction collision is an instance of #4397. A simple workaround is to move the waiting position for the left turns (internal junction) back by 0.5m by setting connection attribute 'contPos' for that movement.

**Comment by heinovski** [comment]
Using the latest version (137ee3ceddd3cd87f2a6eaaef9fc07082bfb4b67), the issue still persists. There are collisions on the intersection (see below) and the rescue vehicle do not use the opposite lane.

> Warning: Vehicle 'from_west.16' performs emergency braking with decel=-9.00 wished=4.50 severity=1.00, time=152.60.
Warning: Vehicle 'from_west.16'; junction collision with vehicle 'case_1_engine', lane=':junc4_17_1', gap=-1.00', latGap=0.00, time=152.60 stage=move.

**Comment by namdre** [comment]
The collision comes down to a lack of lateral spacing while going around the curve. In your test you can work around this by setting lcPushy=0. A possible fix would be to disable pushy behavior automatically when turning at an intersection.

Opposite driving through turns also isn't working but that should be fixable.

**Comment by namdre** [comment]
The simulation looks much better when increasing the width of the roads to 3.84m (one more sublane)

**Comment by namdre** [comment]
@heinovski A number of collisions are now fixed via #9919

**Comment by namdre** [comment]
Since this ticket is growing a bit unwieldy after several iterations of fixes, please open up a new issue with fresh test cases as needed.
